File: 0507-perfect-number/0507-perfect-number.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        long long nim=num;
        long long sum=1;
        if(nim<=1){
            return false;}
        for(int i=2;i<sqrt(nim);i++){
            if(nim%i==0){
                sum=sum+i;
                if(i!=nim/i)
                    sum+=nim/i;     
    }}return sum==nim;
}};

File: 0507-perfect-number/0507-perfect-number.py
class Solution(object):
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        sum=1
        if num<=1:
            return False
        for i in range(2,int(num**0.5)+1):
            if num%i==0:
                sum=sum+i
                if i!=num//i:
                    sum+=num//i     
        return sum==num
